spec.html

```diff
@@ -42358,28 +42468,28 @@ <h1>
           GetWaiterList (
             _block_: a Shared Data Block,
             _i_: a non-negative integer that is evenly divisible by 4,
-          ): a WaiterList
+          ): a WaiterList Record
         </h1>
         <dl class="header">
         </dl>
         <emu-alg>
           1. Assert: _i_ and _i_ + 3 are valid byte offsets within the memory of _block_.
-          1. Return the WaiterList that is referenced by the pair (_block_, _i_).
+          1. Return the WaiterList Record that is referenced by the pair (_block_, _i_).
         </emu-alg>
       </emu-clause>
 
       <emu-clause id="sec-entercriticalsection" type="abstract operation">
         <h1>
           EnterCriticalSection (
-            _WL_: a WaiterList,
+            _WL_: a WaiterList Record,
           ): ~unused~
         </h1>
         <dl class="header">
         </dl>
         <emu-alg>
-          1. Assert: The surrounding agent is not in the critical section for any WaiterList.
+          1. Assert: The surrounding agent is not in the critical section for any WaiterList Record.
           1. Wait until no agent is in the critical section for _WL_, then enter the critical section for _WL_ (without allowing any other agent to enter).
-          1. If _WL_ has a Synchronize event, then
+          1. If _WL_.[[MostRecentLeaveEvent]] is ~empty~, then
```

Comment:
```suggestion
          1. If _WL_.[[MostRecentLeaveEvent]] is not ~empty~, then
```

Also, 6 lines down, there's a reference to
`the Synchronize event in _WL_`
which should probably be
`_WL_.[[MostRecentLeaveEvent]]`